src/devices/Display/LedSegmentDisplay5641AS.cs

```diff
@@ -0,0 +1,190 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Iot.Device.Display;
+
+namespace Display
+{
+    /// <summary>
+    /// LED 4-digit 7-segment display, model 5641AS. This class handles driverless, direct-pin access over GPIO.
+    /// </summary>
+    /// <remarks>
+    /// Data Sheet:
+    /// http://www.xlitx.com/datasheet/5641AS.pdf
+    /// </remarks>
+    public class LedSegmentDisplay5641AS : ISevenSegmentDisplay, IDisposable
+    {
+        private readonly LedSegmentDisplay5641ASPinScheme _pinScheme;
+        private readonly GpioController _gpioController;
+        private readonly bool _shouldDispose;
+        private bool _disposedValue;
+
+        private List<GpioPin> _segments;
+        private List<GpioPin> _digits;
+
+        /// <inheritdoc />
+        public int NumberOfDigits => _pinScheme.DigitCount;
+
+        /// <inheritdoc />
+        public Segment this[int address] { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LedSegmentDisplay5641AS"/> class.
+        /// </summary>
+        public LedSegmentDisplay5641AS(LedSegmentDisplay5641ASPinScheme pinScheme, GpioController? gpioController = null, bool shouldDispose = true)
+        {
+            _pinScheme = pinScheme;
+            _gpioController = gpioController ?? new GpioController();
+            _shouldDispose = gpioController is null || shouldDispose;
+
+            _segments = new(pinScheme.SegmentCountPerDigit);
+            _digits = new(pinScheme.DigitCount);
+        }
+
+        /// <summary>
+        /// Opens GPIO pins defined in <see cref="LedSegmentDisplay5641ASPinScheme"/> as <see cref="PinMode.Output"/>
+        /// </summary>
+        public void Open()
```

Comment:
Why do you have the Open and Close functions? Can Open be done in the constructor and Close in the Dispose? So it's really fully transparent for the user? 
You can keep the function and use them but then change the visibility in private in that case.
I may have miss something why you need an Open and Close as public

Reply:
No reason, just carryover from testing. I will make the changes.